src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -550,6 +587,27 @@ DiagnosticBag analyze(bool strictAnalysis)
                 {
                     bool badRegion = false;
                     walker.Analyze(ref badRegion, result);
+                    if (walker._implicitlyInitializedFieldsOpt is { } implicitlyInitializedFields)
+                    {
+                        Debug.Assert(walker._requireOutParamsAssigned);
+                        var builder = ArrayBuilder<FieldSymbol>.GetInstance(implicitlyInitializedFields.Count);
+                        // ensure the fields to initialize are ordered deterministically
```

Comment:
>ensure the fields to initialize are ordered deterministically [](http://example.com/codeflow?start=27&length=61)

It looks like the loop is O(n^2) where n is the number of implicitly initialized fields. Would it make sense to use `implicitlyInitializedFields.Sort(LexicalOrderSymbolComparer.Instance)` instead.

Reply:
I don't think this implementation is O(n^2) because we are checking 'Contains' on a set. The lexical comparer is probably faster, though, and it will get a deterministic result, which is all we need here.

Reply:
Have changed to use LexicalOrderSymbolComparer.